### DISCUSSION TITLE
Modifying datamodel (adding a new list to dataroot)

### DISCUSSION THREAD

**Discussion by HansITChange** [microstream-one/microstream#296] [Q&A] [answered]
Hello, I've also asked this question on StackOverflow, but as I need a response very quickly, I'm trying this possibility as well.

I have a fully operational application on top of Microstream (with 'Customer' objects). Now I need to add a completely new class 'Vendors', but I'm stuck.
Extending my dataroot class with a list of Vendors, gives me a null list when the database starts (obvious, because there is no vendor in the database yet). A new vendor cannot be added to the list of Vendors, as that would be adding an element to a null list. With a List<Vendor> newvendors = new ArrayList<>(), I can store this. However, when restarting the application, the new object does not appear in the database.
Can somebody explain me how I should extend my datamodel ?

**Comment by fh-ms** [comment, marked as answer]
Hi, you are right, the `vendors` list is not present in the storage, so the field will be initialized with its default value (null).
There are several possibilities to introduce initial values to new fields.
One rather complex way would be to implement a [Legacy Type Handler](https://github.com/microstream-one/microstream/tree/master/examples/custom-legacy-type-handler/src/main/java/one/microstream/examples/customlegacytypehandler).
A far more simple one is just lazy initialization in your `Customer` type:
```java
public List<Vendor> getVendors()
{
    if(this.vendors == null)
    {
        this.vendors = new ArrayList<>()
    }
    return this.vendors;
}
```

**Reply by HansITChange** [reply]
This is a simple, and working solution. Thanks !
